### PR TITLE
[UI/cypress] Add integration test for volume page

### PR DESCRIPTION
**Pull request by ChengYanJin**
**Component**: ui, tests

**Context**: 
Add navigation test for the Volume Page.
Should be merged to integration branch `feature/2878-integration-tests`

Note that:
The first three commits belong to another unmerged PR bugfix the auto-selection of the volume list.

**Summary**

**Acceptance criteria**: 
Pass the test

### CODE REVIEW

File: ui/cypress/support/commands.js [hunk 1]
```diff
@@ -56,6 +56,18 @@ Cypress.Commands.add('setupMocks', () => {
     'api/prometheus/api/v1/query?query=node_uname_info',
     'fixture:prometheus/node-uname-info.json',
   ).as('getNodeUNameInfo');
+  cy.route(
+    'api/alertmanager/api/v2/alerts',
+    'fixture:alertmanager/alerts.json',
+  ).as('getAlerts');
```
Comment (gdemonet) [inline ui/cypress/support/commands.js]: You'll likely need to rebase to make sure this doesn't conflict with what was already merged: https://github.com/scality/metalk8s/pull/2891/files#diff-c23023a5b2e52a3d6183689d6a32d9411ba3559297718fddd6068ee55e822875R143-R144

File: ui/cypress/integration/volume/volumelist.spec.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  cy.setupMocks();
+  cy.login();
+});
+
+// Navigation tests
+describe('Volume list', () => {
+  beforeEach(() => {
+    // Visit is automatically prefixed with baseUrl
+    cy.visit('/volumes');
+    cy.stubHistory();
```
Comment (gdemonet) [inline ui/cypress/integration/volume/volumelist.spec.js]: If you `visit()` in `beforeEach`, make sure you don't `visit()` again within each test case, or it may create issues with the test (some network requests would be interrupted) - see [`07a34b5` (#2891)](https://github.com/scality/metalk8s/pull/2891/commits/07a34b5af5df07fbfa95f08408a93f3fff7eb043)

File: ui/cypress/integration/volume/volumelist.spec.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  });
+
+  it('brings me to the overview tab of the unhealthy volume', () => {
+    // Volume `master-0-alertmanager` has alert.
+    // According to the default sorting rule, it should appear at the first place.
+    cy.wait(['@getPVs', '@getPVCs', '@getAlerts', '@getVolumes']);
```
Comment (gdemonet) [inline ui/cypress/integration/volume/volumelist.spec.js]: Do we really need to wait for those? If we do, you'll want to rebase and make sure aliases are available as you expect - see [`ffef869` (#2891)](https://github.com/scality/metalk8s/pull/2891/commits/ffef8699ccd12ed6a15837f7cb5b72222ca46978)

File: ui/cypress/integration/volume/volumelist.spec.js [hunk 4]
```diff
@@ -38,7 +38,7 @@ describe('Volume list', () => {
     cy.get('[data-cy="volume_table_name_cell"]')
       .contains('prom-m0-reldev')
       .click();
-    cy.get('@historyPush').should('be.calledOnce').and('be.calledWithExactly', {
+    cy.get('@historyPush').should('be.calledWithExactly', {
```
Comment (gdemonet) [inline ui/cypress/integration/volume/volumelist.spec.js]: An issue with the stub catching more than one call? Likely caused by the multiple calls to `cy.visit` :wink:

File: ui/src/containers/VolumePageContent.js [hunk 5]
```diff
@@ -180,19 +180,21 @@ const VolumePageContent = (props) => {
...
         ),
+      'data-cy': 'overview_tab_volume_page',
     },
     {
       selected: isAlertsPage,
       title: (
         <span>
           {intl.translate('alerts')}
-          {PVCName && <TextBadge>{alertlist?.length}</TextBadge>}
+          <TextBadge>{alertlist?.length}</TextBadge>
```
Comment (gdemonet) [inline ui/src/containers/VolumePageContent.js]: Is this related to tests in some way?

Comment (ChengYanJin) [reply ui/src/containers/VolumePageContent.js]: Good catch! Thank you very much! 
I should not make the change here.
